### PR TITLE
Fix TableScan with dynamic filter on partition key pushed down from the join

### LINKED ISSUE #4227: Get unexpect result when join with partition key
### Bug description

Here is my UT.
```TEST_F(HashJoinTest, table) {
  dwio::common::registerReaderFactory(
      std::make_shared<facebook::velox::parquet::ParquetReaderFactory>());
  auto filePath = facebook::velox::test::getDataFilePath(
      "",
      "part-00000-38393452-669f-4a02-b870-f88d178d81ba.c000.snappy.parquet");
  auto split = facebook::velox::exec::test::HiveConnectorSplitBuilder(filePath)
                   .fileFormat(dwio::common::FileFormat::PARQUET)
                   .start(0)
                   .length(472)
                   .partitionKey("k", "0")
                   .build();
  auto outputType = ROW({"n1_0", "n1_1"}, {BIGINT(), BIGINT()});
  std::unordered_map<std::string, std::shared_ptr<connector::ColumnHandle>>
      assignments;
  std::shared_ptr<connector::hive::HiveTableHandle> tableHandle =
      std::make_shared<connector::hive::HiveTableHandle>(
          "test-hive",
          "hive_table",
          true,
          connector::hive::SubfieldFilters{},
          nullptr);
  assignments["n1_0"] = std::make_shared<connector::hive::HiveColumnHandle>(
      "id", connector::hive::HiveColumnHandle::ColumnType::kRegular, BIGINT());
  assignments["n1_1"] = std::make_shared<connector::hive::HiveColumnHandle>(
      "k",
      connector::hive::HiveColumnHandle::ColumnType::kPartitionKey,
      BIGINT());
  std::vector<RowVectorPtr> buildVectors{
      makeRowVector({"c0"}, {makeFlatVector<int64_t>({0, 1, 2})})};
  createDuckDbTable("t", buildVectors);
  core::PlanNodeId probeScanId;
  auto planNodeIdGenerator = std::make_shared<core::PlanNodeIdGenerator>();
  auto op =
      PlanBuilder(planNodeIdGenerator)
          .tableScan(outputType, tableHandle, assignments)
          .capturePlanNodeId(probeScanId)
          .hashJoin(
              {"n1_1"},
              {"c0"},
              PlanBuilder(planNodeIdGenerator).values(buildVectors).planNode(),
              "",
              {"c0", "n1_0", "n1_1"},
              core::JoinType::kInner)
          .project({"c0"})
          .planNode();
  std::cout << "plan node :" << op->toString(true, true);
  SplitInput splits = {{probeScanId, {exec::Split(split)}}};

  HashJoinBuilder(*pool_, duckDbQueryRunner_, driverExecutor_.get())
      .planNode(std::move(op))
      .inputSplits(splits)
      .referenceQuery("")
      .run();
}
```

[Expected behavior]:
This plan should return 0.
"part-00000-38393452-669f-4a02-b870-f88d178d81ba.c000.snappy.parquet" is very simple file. it has only 1  data.
Here is a screen shot. when I use spark to read this parquet.
image

[actual behavior].
This plan return nothing.

### System information

Velox System Info v0.0.2
Commit: 9756de9c4c6b251cc0864e26ea3bf79d32a58cde
CMake Version: 3.16.3
System: Linux-5.4.0-128-generic
Arch: x86_64
C++ Compiler: /usr/bin/c++
C++ Compiler Version: 9.4.0
C Compiler: /usr/bin/cc
C Compiler Version: 9.4.0
CMake Prefix Path: /usr/local;/usr;/;/usr;/usr/local;/usr/X11R6;/usr/pkg;/opt

Conda Env
<details>

```
# packages in environment at /mnt/DP_disk1/zzh/miniconda3:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main    defaults
_openmp_mutex             4.5                       1_gnu    defaults
alabaster                 0.7.13                   pypi_0    pypi
babel                     2.11.0                   pypi_0    pypi
brotlipy                  0.7.0           py39h27cfd23_1003    defaults
ca-certificates           2022.3.29            h06a4308_1    defaults
certifi                   2021.10.8        py39h06a4308_2    defaults
cffi                      1.15.0           py39hd667e15_1    defaults
charset-normalizer        2.0.4              pyhd3eb1b0_0    defaults
click                     8.1.3                    pypi_0    pypi
colorama                  0.4.4              pyhd3eb1b0_0    defaults
conda                     4.12.0           py39h06a4308_0    defaults
conda-content-trust       0.1.1              pyhd3eb1b0_0    defaults
conda-package-handling    1.8.1            py39h7f8727e_0    defaults
cryptography              36.0.0           py39h9ce1e76_0    defaults
docutils                  0.19                     pypi_0    pypi
duckdb                    0.6.2.dev623             pypi_0    pypi
idna                      3.3                pyhd3eb1b0_0    defaults
imagesize                 1.4.1                    pypi_0    pypi
importlib-metadata        6.0.0                    pypi_0    pypi
jdot                      0.5                      pypi_0    pypi
jinja2                    3.1.2                    pypi_0    pypi
ld_impl_linux-64          2.35.1               h7274673_9    defaults
libffi                    3.3                  he6710b0_2    defaults
libgcc-ng                 9.3.0               h5101ec6_17    defaults
libgomp                   9.3.0               h5101ec6_17    defaults
libstdcxx-ng              9.3.0               hd4cf53a_17    defaults
markupsafe                2.1.1                    pypi_0    pypi
ncurses                   6.3                  h7f8727e_2    defaults
numpy                     1.23.5                   pypi_0    pypi
openssl                   1.1.1n               h7f8727e_0    defaults
packaging                 23.0                     pypi_0    pypi
pip                       21.2.4           py39h06a4308_0    defaults
protobuf                  3.20.2                   pypi_0    pypi
pycosat                   0.6.3            py39h27cfd23_0    defaults
pycparser                 2.21               pyhd3eb1b0_0    defaults
pygments                  2.14.0                   pypi_0    pypi
pyopenssl                 22.0.0             pyhd3eb1b0_0    defaults
pysocks                   1.7.1            py39h06a4308_0    defaults
python                    3.9.12               h12debd9_0    defaults
pytz                      2022.7                   pypi_0    pypi
pyyaml                    6.0                      pypi_0    pypi
readline                  8.1.2                h7f8727e_1    defaults
regex                     2022.10.31               pypi_0    pypi
requests                  2.27.1             pyhd3eb1b0_0    defaults
ruamel_yaml               0.15.100         py39h27cfd23_0    defaults
setuptools                61.2.0           py39h06a4308_0    defaults
six                       1.16.0             pyhd3eb1b0_1    defaults
snowballstemmer           2.2.0                    pypi_0    pypi
sphinx                    6.1.3                    pypi_0    pypi
sphinx-rtd-theme          0.5.1                    pypi_0    pypi
sphinxcontrib-applehelp   1.0.3                    pypi_0    pypi
sphinxcontrib-devhelp     1.0.2                    pypi_0    pypi
sphinxcontrib-htmlhelp    2.0.0                    pypi_0    pypi
sphinxcontrib-jsmath      1.0.1                    pypi_0    pypi
sphinxcontrib-qthelp      1.0.3                    pypi_0    pypi
sphinxcontrib-serializinghtml 1.1.5                    pypi_0    pypi
sqlite                    3.38.2               hc218d9a_0    defaults
substrait-validator       0.0.8                    pypi_0    pypi
tk                        8.6.11               h1ccaba5_0    defaults
tqdm                      4.63.0             pyhd3eb1b0_0    defaults
tzdata                    2022a                hda174b7_0    defaults
urllib3                   1.26.8             pyhd3eb1b0_0    defaults
wheel                     0.37.1             pyhd3eb1b0_0    defaults
xz                        5.2.5                h7b6447c_0    defaults
yaml                      0.2.5                h7b6447c_0    defaults
zipp                      3.11.0                   pypi_0    pypi
zlib                      1.2.12               h7f8727e_1    defaults
```

</details>

### Relevant logs

_No response_

### PR DESCRIPTION
**Pull request by zhejiangxiaomai**
When join key is a partition key, the join may pushdown a filter on partition
key to TableScan. HiveConnector didn't handle this correctly.

```
SELECT *
FROM t, u
WHERE t.partition_key = u.key
```

Fixes #4227

### CODE REVIEW

**Comment by mbasmanova** [comment]
@zhejiangxiaomai Would you update PR description to explain the problem and the solution. Please, add a test.

**Comment by zhejiangxiaomai** [comment]
> @zhejiangxiaomai Would you update PR description to explain the problem and the solution. Please, add a test.

Yes, I am adding test.

File: velox/exec/tests/HashJoinTest.cpp [hunk 1]
```diff
@@ -4252,4 +4254,60 @@ DEBUG_ONLY_TEST_F(HashJoinTest, buildReservationReleaseCheck) {
...
+              PlanBuilder(planNodeIdGenerator).values(buildVectors).planNode(),
+              "",
+              {"c0", "n1_0", "n1_1"},
+              core::JoinType::kInner)
+          .project({"c0"})
+          .planNode();
+
+  SplitInput splits = {{probeScanId, {exec::Split(split)}}};
+
+  HashJoinBuilder(*pool_, duckDbQueryRunner_, driverExecutor_.get())
```
Comment (Yuhta) [inline velox/exec/tests/HashJoinTest.cpp:4306]: Should we put some simple validation for the result?

File: velox/connectors/hive/HiveConnector.cpp [hunk 2]
```diff
@@ -499,7 +495,15 @@ void HiveDataSource::addSplit(std::shared_ptr<ConnectorSplit> split) {
         INTEGER(),
         velox::variant(split_->tableBucketNumber.value()));
   }
+  scanSpec_->resetCachedValues();
```
[the remark this answers is not in the document]
Comment (zhejiangxiaomai) [reply velox/connectors/hive/HiveConnector.cpp:498]: Yes, we need this. because setPartitionValue(childSpec, entry.first, entry.second); will update the child of scanSpec_.

File: velox/exec/tests/HashJoinTest.cpp (hunk 1 above)
Comment (zhejiangxiaomai) [reply velox/exec/tests/HashJoinTest.cpp:4306]: Sorry, I didn't quite understand your comments.  Could you give me a reference?
In my opinion, The result of this plan  is "0";
left table has one column c0 [0, 1, 2]
right table is id [0], k[0].
So I use ```select * from t where t.c0 = 0``` to verify the result.

**Review by Yuhta — COMMENTED** [review]
There are some build failure to be addressed

**Comment by zhejiangxiaomai** [comment]
> There are some build failure to be addressed

Thank you, I'm fixing.

**Comment by mbasmanova** [comment]
@zhejiangxiaomai @Yuhta It looks like this issue is specific to HiveConnector used in TableScan. Can it be reproduced using a query without a join, just table scan + some filters?

**Comment by zhejiangxiaomai** [comment]
Just changed hash join scan to scan+filter test.

File: .circleci/config.yml [hunk 3]
```diff
@@ -260,7 +260,7 @@ jobs:
       - run:
           name: "Build"
           command: |
-            make debug NUM_THREADS=16 MAX_HIGH_MEM_JOBS=8 MAX_LINK_JOBS=8 EXTRA_CMAKE_FLAGS="-DVELOX_ENABLE_ARROW=ON"
+            make debug NUM_THREADS=16 MAX_HIGH_MEM_JOBS=8 MAX_LINK_JOBS=8 EXTRA_CMAKE_FLAGS="-DVELOX_ENABLE_ARROW=ON -DVELOX_ENABLE_PARQUET=ON"
```
Comment (mbasmanova) [inline .circleci/config.yml:263]: Unintended change. Please, remove.

File: velox/common/base/Exceptions.h [hunk 4]
```diff
@@ -16,12 +16,23 @@
 
 #pragma once
 
+#ifdef ARROW_USE_GLOG
```
Comment (mbasmanova) [inline velox/common/base/Exceptions.h:19]: Seems like unrelated change. Please, remove.

File: velox/exec/tests/TableScanTest.cpp [hunk 5]
```diff
@@ -2467,3 +2468,35 @@ TEST_F(TableScanTest, parallelPrepare) {
...
+  auto filePath = facebook::velox::test::getDataFilePath(
+      "velox/exec/tests", "data/bigint.parquet");
+  auto split = HiveConnectorSplitBuilder(filePath)
+                   .fileFormat(dwio::common::FileFormat::PARQUET)
+                   .start(0)
+                   .length(472)
+                   .partitionKey("k", "0")
+                   .build();
+  auto outputType = ROW({"n1_0", "n1_1"}, {BIGINT(), BIGINT()});
+  auto subfieldFilters = singleSubfieldFilter("k", lessThanOrEqual(2));
```
Comment (mbasmanova) [inline velox/exec/tests/TableScanTest.cpp:2484]: Does this issue happen only when we have a subfield-filter on a partition key? If that's the case, then it is a bug in the query planner. Such filters should not be pushed down into table scan, but rather processed during split enumeration. We can add a check to TableScanNode constructor to reject such filters.

**Review by mbasmanova — CHANGES_REQUESTED** [review]

File: velox/exec/tests/TableScanTest.cpp (hunk 5 above)
Comment (zhejiangxiaomai) [reply velox/exec/tests/TableScanTest.cpp:2484]: Yes, this is the case, a subfield-filter on a partition key. Ok I can add a check at TableScanNode  constructor.

File: .circleci/config.yml (hunk 3 above)
Comment (zhejiangxiaomai) [reply .circleci/config.yml:263]: I add it because I add a velox_dwio_parquet_reader for velox_exec_test.

File: velox/exec/tests/TableScanTest.cpp (hunk 5 above)
Comment (mbasmanova) [reply velox/exec/tests/TableScanTest.cpp:2484]: @zhejiangxiaomai This is probably a valid use case. If so, then we should keep these filters separately, not include these in ScanSpec, and evaluate these without creating any file reader by just looking at the partition key values in the split.

File: .circleci/config.yml (hunk 3 above)
Comment (mbasmanova) [reply .circleci/config.yml:263]: You should not do that. The problem here is not specific to file format, hence, can be reproduced with a DWRF file without introducing a dependency on Parquet.

Comment (zhejiangxiaomai) [reply .circleci/config.yml:263]: Thanks, I will use DWRF file.

File: velox/common/base/Exceptions.h (hunk 4 above)
Comment (zhejiangxiaomai) [reply velox/common/base/Exceptions.h:19]: will remove it.

File: velox/exec/tests/TableScanTest.cpp (hunk 5 above)
Comment (zhejiangxiaomai) [reply velox/exec/tests/TableScanTest.cpp:2484]: Here is a simple case in spark sql UT. https://github.com/apache/spark/blob/master/sql/core/src/test/scala/org/apache/spark/sql/DynamicPartitionPruningSuite.scala#L1332-L1359

File: .circleci/config.yml (hunk 3 above)
Comment (mbasmanova) [reply .circleci/config.yml:263]: Thanks. Make sure to write the test case to auto-generate the file, don't check in the file itself. Looks for other tests in TableScanTest for examples.

File: velox/exec/tests/TableScanTest.cpp (hunk 5 above)
Comment (mbasmanova) [reply velox/exec/tests/TableScanTest.cpp:2484]: @zhejiangxiaomai Thanks. This looks like a join on partition key like the example above.

Comment (rui-mo) [reply velox/exec/tests/TableScanTest.cpp:2484]: > Does this issue happen only when we have a subfield-filter on a partition key? If that's the case, then it is a bug in the query planner. Such filters should not be pushed down into table scan, but rather processed during split enumeration. We can add a check to TableScanNode constructor to reject such filters.

Hi @mbasmanova, I believe there are two levels of filter in TableScan, the subfield-filter and the remaining-filter. And we have implemented certain logic to separated the filter conditions in TableScan into these two parts. In above case, the filter condition is created as subfield-filter. Should we change it to remaining-filter or prevent it in TableScan?

Comment (mbasmanova) [reply velox/exec/tests/TableScanTest.cpp:2484]: @rui-mo It would be more efficient to handle subfield filters on a partition keys in the split generation logic by dropping the splits.

**Comment by zhejiangxiaomai** [comment]
Hi @mbasmanova,  I avoid pushing down filter into ScanSpec when we construct ScanSpec.

File: velox/exec/tests/TableScanTest.cpp (hunk 5 above)
Comment (rui-mo) [reply velox/exec/tests/TableScanTest.cpp]: Thank you, I see. Maybe we need to follow the code here to skip the split if it does not satisfy the filter. Is that correct?

https://github.com/facebookincubator/velox/blob/main/velox/connectors/hive/HiveConnector.cpp#L447-L452

cc @@zhejiangxiaomai

File: velox/connectors/hive/HiveConnector.cpp [hunk 6]
```diff
@@ -182,10 +182,14 @@ std::shared_ptr<common::ScanSpec> HiveDataSource::makeScanSpec(
     const std::vector<const HiveColumnHandle*>& columnHandles,
     memory::MemoryPool* pool) {
   auto spec = std::make_shared<common::ScanSpec>("root");
+  std::set<std::string> partitionKeyNameSet;
```
Comment (Yuhta) [inline velox/connectors/hive/HiveConnector.cpp:185]: `folly::FastSet`

File: velox/connectors/hive/HiveConnector.cpp [hunk 7]
```diff
@@ -221,6 +225,9 @@ std::shared_ptr<common::ScanSpec> HiveDataSource::makeScanSpec(
     if (pair.first.toString() == kPath || pair.first.toString() == kBucket) {
       continue;
     }
+    if (partitionKeyNameSet.count(pair.first.toString()) > 0) {
```
Comment (Yuhta) [inline velox/connectors/hive/HiveConnector.cpp:228]: Maybe combine with the previous `if`

Comment (zhejiangxiaomai) [reply velox/connectors/hive/HiveConnector.cpp]: Thanks, just fixed it.

File: velox/connectors/hive/HiveConnector.cpp (hunk 6 above)
Comment (zhejiangxiaomai) [reply velox/connectors/hive/HiveConnector.cpp:185]: thanks, change to  std::unordered_set<std::string> partitionKeys;

**Comment by zhejiangxiaomai** [comment]
Hi, @Yuhta @mbasmanova Would you review again?

I use partitionKeyfilters_ to save filter on partitionKey.  Then,  use testPartitionFilters to skip split by filter.

File: velox/connectors/hive/HiveConnector.cpp [hunk 8]
```diff
@@ -182,10 +182,14 @@ std::shared_ptr<common::ScanSpec> HiveDataSource::makeScanSpec(
     const std::vector<const HiveColumnHandle*>& columnHandles,
     memory::MemoryPool* pool) {
   auto spec = std::make_shared<common::ScanSpec>("root");
+  std::unordered_set<std::string> partitionKeyNameSet;
```
Comment (Yuhta) [inline velox/connectors/hive/HiveConnector.cpp:185]: No need to filter here as `filters` here are all static

File: velox/exec/tests/TableScanTest.cpp [hunk 9]
```diff
@@ -2511,3 +2511,60 @@ TEST_F(TableScanTest, parallelPrepare) {
...
+    auto rowVector = makeRowVector(
+        {makeFlatVector<int64_t>(size, [&](auto row) { return row; })});
+    writeToFile(filePaths[i]->path, rowVector);
+    rowVectors.emplace_back(rowVector);
+    splits.emplace_back(HiveConnectorSplitBuilder(filePaths[i]->path)
+                            .partitionKey("pkey", partitionKeys[i])
+                            .build());
+  }
+
+  auto lessThanTen = singleSubfieldFilter("pkey", lessThan(10));
```
[the remark this answers is not in the document]
Comment (zhejiangxiaomai) [reply velox/exec/tests/TableScanTest.cpp:2536]: Thanks, I'm working on this PR. But it seems difficult to construct a dynamic filter By addDynamicFilter.  I can not find a example from exsiting UT cases.

File: velox/connectors/hive/HiveConnector.cpp [hunk 10]
```diff
@@ -293,6 +310,8 @@ HiveDataSource::HiveDataSource(
       hiveColumnHandles,
       pool_);
 
+  extractParitionFilters(hiveTableHandle->subfieldFilters());
```
[the remark this answers is not in the document]
Comment (zhejiangxiaomai) [reply velox/connectors/hive/HiveConnector.cpp:313]: Thanks, I think I need extract dynamic filter on partition key in `addSplit`.

File: velox/connectors/hive/HiveConnector.cpp [hunk 11]
```diff
@@ -366,6 +385,75 @@ bool testFilters(
...
+  for (auto& kv : partitionKeyValue) {
+    auto it = partitionKeysHandle.find(kv.first);
+    VELOX_CHECK(
+        it != partitionKeysHandle.end(),
+        "ColumnHandle is missing for partition key {}",
+        kv.first);
+    if (!kv.second.has_value() || partitionKeyFilter.count(kv.first) == 0) {
+      continue;
+    }
+    auto columnStats = createFakePartitionKeyColumnStats(
```
[the remark this answers is not in the document]
Comment (zhejiangxiaomai) [reply velox/connectors/hive/HiveConnector.cpp]: Yes, I used applyFilter.

**Comment by zhejiangxiaomai** [comment]
Hi, @Yuhta I fixed all comments.
The main idea of this commit is 
1. set setIsPartitionKey(true) in addDynamicFilter when this scanSpec is partition key.
2. In testFilters, If scanSpec is partition key and it has filter, use apply filter to choose split.

Would you review this pr when you are free?

File: velox/dwio/common/ScanSpec.h [hunk 12]
```diff
@@ -299,6 +299,13 @@ class ScanSpec {
     enableFilterReorder_ = enableFilterReorder;
   }
 
+  void setIsPartitionKey(bool isPartitionKey) {
+    isPartitionKey_ = isPartitionKey;
+  }
+
+  bool isPartitionKey() const {
+    return isPartitionKey_;
+  }
```
Comment (Yuhta) [inline velox/dwio/common/ScanSpec.h:308]: nit: empty line below

File: velox/connectors/hive/HiveConnector.cpp [hunk 13]
```diff
@@ -347,17 +348,66 @@ HiveDataSource::HiveDataSource(
...
+bool applyPartitionFilter(
+    TypeKind kind,
+    std::string partitionValue,
+    common::Filter* filter) {
+  switch (kind) {
+    case TypeKind::BIGINT:
+    case TypeKind::INTEGER:
+    case TypeKind::SMALLINT:
+    case TypeKind::TINYINT: {
+      return applyFilter(*filter, std::stoi(partitionValue));
```
Comment (Yuhta) [inline velox/connectors/hive/HiveConnector.cpp:360]: Maybe just use `folly::to`?

Comment (zhejiangxiaomai) [reply velox/connectors/hive/HiveConnector.cpp]: Thanks, fixed it.

File: velox/dwio/common/ScanSpec.h (hunk 12 above)
Comment (zhejiangxiaomai) [reply velox/dwio/common/ScanSpec.h:308]: Thanks, updated.

File: velox/dwio/common/ScanSpec.h [hunk 14]
```diff
@@ -419,6 +427,8 @@ class ScanSpec {
   // Only take the first maxArrayElementsCount_ elements from each array.
   vector_size_t maxArrayElementsCount_ =
       std::numeric_limits<vector_size_t>::max();
+
+  bool isPartitionKey_ = false;
```
[the remark this answers is not in the document]
Comment (zhejiangxiaomai) [reply velox/dwio/common/ScanSpec.h:431]: Thanks, just updated.

File: velox/connectors/hive/HiveConnector.cpp [hunk 15]
```diff
@@ -347,17 +347,57 @@ HiveDataSource::HiveDataSource(
...
+    std::unordered_map<std::string, std::shared_ptr<HiveColumnHandle>>&
+        partitionKeysHandle) {
   auto totalRows = reader->numberOfRows();
   const auto& fileTypeWithId = reader->typeWithId();
   const auto& rowType = reader->rowType();
   for (const auto& child : scanSpec->children()) {
     if (child->filter()) {
       const auto& name = child->fieldName();
       if (!rowType->containsChild(name)) {
+        if (child->isPartitionKey()) {
```
Comment (Yuhta) [inline velox/connectors/hive/HiveConnector.cpp:392]: Actually I am wondering do we really need to mark it as partition key in `ScanSpec`?  If we can find the `name` in `partitionKey` here, we can already decide that it is a partition key and call `applyPartitionFilter` on it, no?

Comment (zhejiangxiaomai) [reply velox/connectors/hive/HiveConnector.cpp:392]: I think it makes sense. I'll make some changes.

File: velox/connectors/hive/HiveConnector.cpp [hunk 16]
```diff
@@ -347,17 +347,56 @@ HiveDataSource::HiveDataSource(
 }
 
 namespace {
+bool applyPartitionFilter(
+    TypeKind kind,
+    std::string partitionValue,
```
Comment (Yuhta) [inline velox/connectors/hive/HiveConnector.cpp:352]:
```suggestion
    const std::string& partitionValue,
```

File: velox/connectors/hive/HiveConnector.cpp [hunk 17]
```diff
@@ -347,17 +347,56 @@ HiveDataSource::HiveDataSource(
...
+      return applyFilter(*filter, folly::to<double>(partitionValue));
+    }
+    case TypeKind::BOOLEAN: {
+      return applyFilter(*filter, folly::to<bool>(partitionValue));
+    }
+    case TypeKind::VARCHAR: {
+      return applyFilter(*filter, partitionValue);
+    }
+    default:
+      VELOX_CHECK(false, "Bad partitionKey type to filter");
```
Comment (Yuhta) [inline velox/connectors/hive/HiveConnector.cpp:372]:
```suggestion
      VELOX_FAIL("Bad type {} for partition value: {}", kind, partitionValue);
```
